src/arviz_stats/psens.py

```diff
@@ -0,0 +1,213 @@
+        Coordinates defining a subset over the component element for which to
+        compute the prior sensitivity diagnostic.
+    var_names : list of str, optional
+        Names of posterior variables to include in the power scaling sensitivity diagnostic
+    coords : dict, optional
+        Coordinates defining a subset over the posterior. Only these variables will
+        be used when computing the prior sensitivity.
+    filter_vars: {None, "like", "regex"}, default None
+        If ``None`` (default), interpret var_names as the real variables names.
+        If "like", interpret var_names as substrings of the real variables names.
+        If "regex", interpret var_names as regular expressions on the real variables names.
+    delta : float
+        Value for finite difference derivative calculation.
+
+
+    Returns
+    -------
+    xarray.DataTree
+        Returns dataTree of power-scaling sensitivity diagnostic values.
+        Higher sensitivity values indicate greater sensitivity.
+        Prior sensitivity above 0.05 indicates informative prior.
+        Likelihood sensitivity below 0.05 indicates weak or non-informative likelihood.
+
+    Notes
+    -----
+    The diagnostic is computed by power-scaling the specified component (prior or likelihood)
+    and determining the degree to which the posterior changes as described in [1]_.
+    It uses Pareto-smoothed importance sampling to avoid refitting the model.
+
+    References
+    ----------
+    .. [1] Kallioinen et al, *Detecting and diagnosing prior and likelihood sensitivity with
+       power-scaling*, 2022, https://arxiv.org/abs/2107.14054
+    """
+    dataset = extract(dt, var_names=var_names, filter_vars=filter_vars, group="posterior")
+    if coords is None:
+        dataset = dataset.sel(coords)
+
+    if group == "log_likelihood":
+        component_draws = get_log_likelihood(dt, var_name=component_var_names, single_var=False)
+    elif group == "log_prior":
+        component_draws = get_log_prior(dt, var_names=component_var_names)
+    else:
+        raise ValueError("Value for `group` argument not recognized")
+
+    component_draws = component_draws.stack(__sample__=("chain", "draw"))
+    if component_coords is None:
+        component_draws = component_draws.sel(component_coords)
+    if isinstance(component_draws, xr.DataArray):
+        component_draws = component_draws.to_dataset()
+    if len(component_draws.dims):
+        component_draws = component_draws.to_stacked_array(
+            "latent-obs_var", sample_dims=("__sample__",)
+        ).sum("latent-obs_var")
+
+    component_draws = component_draws.unstack()
+
+    # calculate lower and upper alpha values
+    lower_alpha = 1 / (1 + delta)
+    upper_alpha = 1 + delta
+
+    # calculate importance sampling weights for lower and upper alpha power-scaling
+    lower_w = np.exp(component_draws.azstats.power_scale_lw(alpha=lower_alpha)).values.flatten()
+    lower_w = lower_w / np.sum(lower_w)
```

Comment:
I have always been a bit shocked by this. Asssuming `lower_w` comes from the log prior and is squished into a 2d array of shape `(n_variables, n_samples)` do we really want to sum over both dimensions?

Reply:
I had missed the `.sum("latent-obs_var")` piece. So we sum over sample dims only. We have already summed over all variables and dims.